lib/galaxy_test/api/test_users.py

```diff
@@ -129,6 +134,45 @@ def test_undelete_user(self):
         undeleted_user = self._get(f"users/{user['id']}", admin=True).json()
         assert undeleted_user["deleted"] is False, undeleted_user
 
+    @requires_admin
+    @requires_new_user
+    @requires_new_history
+    @skip_without_tool("cat_data_and_sleep")
+    def test_delete_user_cancel_all_jobs(self):
+        dataset_populator = DatasetPopulator(self.galaxy_interactor)
+        with self._different_user(TEST_USER_EMAIL_DELETE_CANCEL_JOBS):
+            user_id = self._get_current_user_id()
+            history_id = dataset_populator.new_history()
+            hda_id = dataset_populator.new_dataset(history_id)["id"]
+
+            inputs = {
+                "input1": {"src": "hda", "id": hda_id},
+                "sleep_time": 6000,
+            }
+            run_response = dataset_populator.run_tool_raw(
+                "cat_data_and_sleep",
+                inputs,
+                history_id,
+            )
+            self._assert_status_code_is_ok(run_response)
+
+            job_id = run_response.json()["outputs"][0]["id"]
```

Comment:
ahhhh, that's not the job id 😆

Reply:
Ouch! :sweat_smile:  Thanks for spotting it!